### PR TITLE
Use window.location to retrieve the username for GraphQL variables

### PR DESCRIPTION
**Pull request by kinow**
This is a small change with no associated Issue.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

File: src/mixins/graphql.js [hunk 1]
```diff
@@ -51,7 +52,8 @@ export default {
      * @return {string} - the Workflow ID used in this view
      */
     workflowId () {
-      return `${this.user.username}|${this.workflowName}`
+      const username = getUserNameFromUrl(this.user.username)
+      return `${username}|${this.workflowName}`
```
Comment (kinow) [inline src/mixins/graphql.js:56]: The Vuex user is still there, but we will try to use the user found in the URL. If no user found, then we fallback to the Vuex value.

File: src/utils/urls.js [hunk 2]
```diff
@@ -72,6 +72,28 @@ function createUrl (path, websockets = false, baseOnly = false) {
...
+ *
+ * The expected URL format is <protocol>://<hostname>:<port>/user/<username>/. The returned
+ * value will be either the <username> or the given fallback username.
+ *
+ * @param {string} fallbackValue - fallback value
+ * @returns {string} - the
+ */
+function getUserNameFromUrl (fallbackValue) {
+  return window.location.pathname
+    ? window.location.pathname
```
Comment (kinow) [inline src/utils/urls.js:88]: TodayILearned it's better/safer to use `window.location` than `document.location`, even though I get the same with both on Firefox/Chromium/etc: https://stackoverflow.com/a/2431375

**Comment by kinow** [comment]
Just need to prepare the set-up to test @datamel 's PR again, to validate that this PR indeed allows us to access the workflows of other users.... then it will be ready for review.

**Comment by kinow** [comment]
It works hohoho!

image

Left monitor with `cylc` user, but looking at `kinow|five`. Right monitor with `kinow` user looking at `kinow|five`.

Using the latest commit from @datamel 's https://github.com/cylc/cylc-uiserver/pull/204

Also tested changing 

```py
c.UIServer.user_authorization = {
  "cylc": ["ALL"]                                 
}
```

to

```py
c.UIServer.user_authorization = {
  "cylc": ["!ALL"]                                 
}
```

in `~/.cylc/hub/config.py` and got the 403 error as expected :+1: so ready for review, shouldn't break the UI if merged now, but can be used to help reviewing the UIS PR.

**Review by hjoliver — APPROVED** [review]

**Comment by kinow** [comment]
Note: @oliver-sanders ' #711 PR uses the owner property returned by the UIS. This owner would fix the issue here too, so we might hold back on merging this one until @datamel finishes reviewing that PR. If merged, I'll test it again, and close this PR and the associated issue if the issue doesn't occur anymore 👍

**Comment by kinow** [comment]
Superseded by #711

**Pull request closed without merging**